core/src/main/java/org/mule/routing/outbound/AbstractMessageSequenceSplitter.java

```diff
@@ -50,6 +51,7 @@
     protected MessageInfoMapping messageInfoMapping;
     protected int batchSize;
     protected String counterVariableName;
+    private static AtomicInteger processedSequences = new AtomicInteger(0);
```

Comment:
does this need to be at the class level and atomic? Can't it be scoped to the `process` method.